Main interface features

RefactorForge opened the pull request:
Impelementation of main interface with functions contorling flow of the application, entry main point and some basic test cases.

Kindly asking for review

Thank you

Commit:
feat: Adds implementation of flow_functions module and main entry point for appliction
	- Implemented functions for application workflow control and management
	- Everything put together in entry point

Commit:
test: Adds tests for flow_function module
	- Testing basic behaviour of control functions that are used in main file
	- Some behaviour is mocked due to keeping tests in unit scope area

Commit:
docs: Add comprehensive README with setup, usage, and pre-commit hook instructions

tynaneu commented on main.py:
```diff
@@ -0,0 +1,26 @@
+            io_handler, args.input, args.sample_weight
+        )
+
+        result = execute_clustering(container, train_data, train_sample_weight)
```
execute clustering can also have just the clustering instance (as io_handler) and not take the whole container - you should be consistent in these things

RefactorForge replied:
Refactored.

tynaneu commented on app/flow_functions.py:
```diff
@@ -0,0 +1,256 @@
+            raw_config = yaml.safe_load(file)
+        validated_config = Config(**raw_config)
+        return validated_config
+    except Exception as e:
```
except only ValidationError not all exceptions. If there was some other exception raised the reraised exception (with the message "Validation Error" would be misleading.

RefactorForge replied:
Refactored.

Commit:
refactor: Changes function, variables name and docs:
	- Implemented changes appointd in review coments
	- Updated naming and usage of io_handlers to correctly reflect changes in dependency module
	- Simplified docs
	- Implemented check and conversion for sample_weight array to reflect correct type in folowing flow

Commit:
refactor: Changes variables and their usage in main:
	- Changed general io_handler to input and output handler and created clustering_algorithm variable to be
	passed into execute_clustering as instance instead as whole container

Commit:
test: Refactors flow_functions test cases:
	- Updated test cases to respect changes in flow_function module

Acnologia7 commented:
@tynaneu 
Should be resolved now.
Kindly asking for review.

Thank you

tynaneu commented on main.py:
```diff
@@ -0,0 +1,32 @@
+        )
+
+        save_results(output_handler, result, args.output)
+    except Exception as e:
```
The function currently uses a generic except Exception block, which suppresses all exceptions, including unexpected ones, making debugging difficult. It should handle specific exceptions where possible and use tools loging stack traces for better debugging. Additionally, re-raising exceptions after logging ensures critical errors aren't silently ignored, maintaining transparency and reliability.

RefactorForge replied:
Refactored.

Commit:
refactor: Updates exception handling through the application and docstrings
	- Implemented logging mechanism in main module
	- Extended exception handling via more specific exception cases in io_handler and flow_functions modules
	- Updated docstrings to respect this change

Acnologia7 commented:
@tynaneu 
Should be resolved now.
Kindly asking for review.

Thank you
